lib/extract_field/credentials_intelligence/CredentialsIntelligenceProtocolFactory.js

```diff
@@ -0,0 +1,20 @@
+const { CIVersion } = require('../../enums/CIVersion');
+const { V1CredentialsIntelligenceProtocol } = require('./V1CredentialsIntelligenceProtocol');
+const { MultistepSSOCredentialsIntelligenceProtocol } = require('./MultistepSSOCredentialsIntelligenceProtocol');
+
+class CredentialsIntelligenceProtocolFactory {
+    static Create(protocolVersion) {
+        switch (protocolVersion) {
+            case CIVersion.V1:
+                return new V1CredentialsIntelligenceProtocol();
+            case CIVersion.MULTISTEP_SSO:
+                return new MultistepSSOCredentialsIntelligenceProtocol();
+            default:
+                throw new Error(`Unknown CI protocol version '${protocolVersion}', acceptable versions are ${Object.values(CIVersion)}`);
```

Comment:
They will use `px_credentials_intelligence_version` in the configuration;  no need to manually edit the code at all. If it's a value other than `v1` or `multistep_sso` it will throw an error on initialization.